### ISSUE TITLE
The Training Loss Doesnt Decrease at All

### ISSUE THREAD

**Issue by christofel04** [Toytiny/RadarOcc#13]
Hello Wang and friends,

great work on 3D occupancy prediction using 4D radar.

Actually I try to train 3D occupancy prediction using 4D radar dataset but my training loss doesnt decrease at all as below:

<img width="1440" height="900" alt="Image" src="https://github.com/user-attachments/assets/8fd1bcb2-9df2-4308-9a12-c6df86d97a8e" />

I am using K-Radar Sparse Tensor using 8 Dopplers generated with code "generated_4d_polar_doppler".

But I got warning error that my input feature doesnt have gradient None as below.

<img width="796" height="273" alt="Image" src="https://github.com/user-attachments/assets/b86f6c65-2099-4e2f-b9e1-f1105de28e75" />

Is this warning error make training 3D occupancy prediction using 4D radar cannot be trained?

**Comment by Ericwen2001** [comment]
Hi christofel04,  I believe this behavior is due to the loss normalization applied during training.
[https://github.com/Toytiny/RadarOcc/blob/1bc1c3f3b1d11aeaa217e2336a0086f70705926b/projects/occ_plugin/occupancy/detectors/radarocc_self_small.py#L506]
 As seen in the code here, loss normalization is commonly used in the community to balance multiple loss functions.
 The decrease in grad_norm suggests that training is progressing as expected. Regarding the warning, it’s hard to determine the cause just from this screenshot—it might be safe to ignore, but additional context would help confirm.

**Comment by christofel04** [comment]
Hello Eric and teams,

thank you for prompt answer.

It make sense because actually I have problem when using Lovasz loss hence I dont use Lovasz loss as below:

<img width="1119" height="359" alt="Image" src="https://github.com/user-attachments/assets/fdfc2aa9-ab31-41c1-a3c7-61f4e03298a1" />

Then I use normal segmentation loss as below in Lovasz function.

<img width="1124" height="670" alt="Image" src="https://github.com/user-attachments/assets/0f5a1f84-4bcf-4d49-8dc4-85ed55bb873a" />

Do you know how to solve problem with Lovasz loss? Because I dont have any problem when installing torch and cuda toolkits.

**Comment by YueSun0609** [comment]
I met the same problem
